dcore/state.py

```diff
@@ -120,6 +121,10 @@ def dump(self):
             self.diagnostic_data = defaultdict(partial(defaultdict, list))
 
         if (next(self.dumpcount) % self.output.dumpfreq) == 0:
+
+            if 'Courant' in self.diagnostic_fields.diagnostic_field_dict.keys():
+                self.diagnostic_fields.Courant(self)
+
```

Comment:
This smells of bad design.  You shouldn't be attempting to enumerate the diagnostic fields here.